### DISCUSSION TITLE
'IImageProcessingContext' does not contain a definition for 'ApplyScalingWaterMark'

### DISCUSSION THREAD

**Discussion by SibeeshVenu** [SixLabors/ImageSharp#1281] [Q&A] [answered]
### Description
I am running the latest version and I am getting the error as preceding.

> Severity	Code	Description	Project	File	Line	Suppression State
> Error	CS1061	'IImageProcessingContext' does not contain a definition for 'ApplyScalingWaterMark' and no accessible extension method 'ApplyScalingWaterMark' accepting a first argument of type 'IImageProcessingContext' could be found (are you missing a using directive or an assembly reference?)	GitHubFuncs	C:\Sibeesh\Github\GitHubFuncs\GetLatestPosts.cs	39	Active

When I run the code below.
```
using(var img = Image.Load("canvas.jpg")) {
    Font font = SystemFonts.CreateFont("Arial", 10);
    using var img2 = img.Clone(ctx => ctx.ApplyScalingWaterMark(font, feedItem.Summary.Text, Color.HotPink, 5, true));
    img2.Save("images/wrapped.png");
}
```

Already added the using statements.

```
using SixLabors.Fonts;
using SixLabors.ImageSharp.Drawing;
using SixLabors.ImageSharp;
using SixLabors.ImageSharp.Processing;
```
What am I missing here? Is this an issue with latest version?

**Comment by JimBobSquarePants** [comment]
There’s no such method. Where did you get the name from?

**Reply by SibeeshVenu** [reply]
Please see here: https://github.com/SixLabors/Samples/blob/master/ImageSharp/DrawWaterMarkOnImage/Program.cs, it was mentioned here: https://github.com/SixLabors/ImageSharp/issues/375#issuecomment-340270912

**Comment by dlemstra** [comment]
Seems to be a custom extension method in this issue: #484?

**Comment by JimBobSquarePants** [comment]
Thanks @dlemstra It'll require updating to `IImageProcessingContext` instead of IImageProcessingContext<TPixel> then. 

Not a bug though @SibeeshVenu , this should have been added to the Help section of our Discussions forum. I'll transfer it now.

**Comment by SibeeshVenu** [comment]
If this is not how we can apply a watermark with a word-wrap feature,  could you please show me how to do this with latest version? Do you have any docs for this?

**Comment by tocsoft** [comment, marked as answer]
This example (which you already linked to) https://github.com/SixLabors/Samples/blob/master/ImageSharp/DrawWaterMarkOnImage/Program.cs works fine on the latest release however you need to include the entire sample into your code base not just the call site as the sample you linked to is specifically demonstrating how to add a custom processing extension that applies the scaling text watermark to an image.

So in conclusion you need to include these lines from the sample https://github.com/SixLabors/Samples/blob/32557bf359066c07358572e7147e638608522503/ImageSharp/DrawWaterMarkOnImage/Program.cs#L43-L159 into your project to get it to actually output.

**Comment by SibeeshVenu** [comment]
Thanks a lot. I got it working!

```
private static string WriteOnImage(SyndicationItem feedItem) {
    using var img = Image.Load("images/canvas.jpg");
    var font = SystemFonts.CreateFont("Arial", 20);
    using var img2 = img.Clone(ctx => ctx.ApplyScalingWaterMark(font, feedItem.Summary.Text, Color.White, 5, true));
    return img2.ToBase64String(PngFormat.Instance);
}
```
